### PR TITLE
Discourage usage of max_t in favour of max_resource_attr (examples, d…

### PR DESCRIPTION
**Pull request by mseeger**
…ocs)

*Issue #, if available:*

*Description of changes:*
- Discourage usage of max_t
- Simplify using max_resource_attr with blackboxes
- Rewrite all examples which used max_t
- Updated docs (also new FAQ entry)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: benchmarking/commons/hpo_main_simulator.py [hunk 1]
```diff
@@ -278,17 +278,19 @@ def main(
...
+                max_resource_attr
             )
             method_kwargs["max_resource_attr"] = max_resource_attr
         else:
-            config_space = trial_backend.blackbox.configuration_space
-            method_kwargs["max_t"] = max_resource_level
+            config_space = blackbox.configuration_space
+            max_resource_value = blackbox.configuration_space_with_max_resource_attr(
+                "BOGUS123"
+            )["BOGUS123"]
```
Comment (wesk) [inline benchmarking/commons/hpo_main_simulator.py:292]: `BOGUS123`?

Comment (mseeger) [reply benchmarking/commons/hpo_main_simulator.py:292]: Just an arbitrary name for max_resource_attr. I can change this.

Comment (mseeger) [reply benchmarking/commons/hpo_main_simulator.py:292]: The idea is that `configuration_space_with_max_resource_attr` knows how to obtain `max_resource_level` from the blackbox, this kind of intransparent expression in the old code.

Comment (mseeger) [reply benchmarking/commons/hpo_main_simulator.py:292]: In fact, let me rewrite this. Let us just avoid using `max_t` here at all.

**Review by wesk — APPROVED** [review]

**Pull request merged**